lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -350,6 +351,11 @@ module InternalConstants
 
     # Whether to call metadata agent to retrieve monitored resource.
     config_param :enable_metadata_agent, :bool, :default => false
+
+    # The url of Metadata Agent. If this config is not customized (left as
+    # default instead) while an environment variable
+    # STACKDRIVER_METADATA_AGENT_URL is set, the value of that variable will
+    # take precedence. Otherwise the value of this config will be used.
```

Comment:
The problem with this approach is that there's no way to distinguish the unset option (where the environment variable should take precedence) and the option that's explicitly set to the default (where the user wants to override the environment variable).
Perhaps we should default this to `nil` and set it to the default value after examining the environment variable if it stays a `nil`?

Reply:
Makes sense. Changed.